### PR TITLE
Chore: upgrade Next.js, clean, and add some features

### PR DESCRIPTION
**Pull request by Mcastres**
This PR:

- Upgrade Next.js to 12
- Remove unnecessary packages
- Replace default article richtext editor with ckeditor 5
- Replace GitHub button with the react one
- Implement SEO on the frontend which was broken
- Dynamically import DZ components on the front-side to increase performance.

### CODE REVIEW

File: api/package.json
Comment (ivanThePleasant) [inline api/package.json:39]: Do we still need node 12 here then?

Comment (Mcastres) [reply api/package.json:39]: Should we set the min to 14?

Comment (ivanThePleasant) [reply api/package.json:39]: Well, Strapi is now officially 14 to 18, and since you've bumped up FoodAdvisor to Strapi 4.5, I think it makes sense to update the node version accordingly. Not that it's gonna make much of a difference or break anything, just following the main product requirements

Comment (Mcastres) [reply api/package.json:39]: Sure! Will do that!

**Comment by ivanThePleasant** [comment]
Ok, I didn't see any issues or anything, and thanks for changing the Node version. LGTM, unless you want to wait for Fabien's review as well 👍

**Review by ivanThePleasant — APPROVED** [review]

**Comment by Mcastres** [comment]
No, it will be enough! Thanks @ivanThePleasant 🙏

**Pull request merged**
